src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -794,6 +794,53 @@ class UseDatabaseStatementSegment(BaseSegment):
     )
 
 
+# Data Manipulation Statements
+@spark3_dialect.segment()
+class InsertTableStatement(BaseSegment):
+    """A `INSERT [TABLE]` statement to insert or overwrite new rows into a table.
+
+    https://spark.apache.org/docs/latest/sql-ref-syntax-dml-insert-into.html
+    https://spark.apache.org/docs/latest/sql-ref-syntax-dml-insert-overwrite-table.html
+    """
+
+    type = "insert_table_statement"
+
+    match_grammar = Sequence(
+        "INSERT",
+        OneOf("INTO", "OVERWRITE"),
+        Ref.keyword("TABLE", optional=True),
+        Ref("TableReferenceSegment"),
+        Ref("PartitionSpecGrammar", optional=True),
+        Ref("BracketedColumnReferenceListGrammar", optional=True),
+        OneOf(
+            AnyNumberOf(
+                Ref("ValuesClauseSegment"),
+                min_times=1,
+            ),
+            Ref("SelectableGrammar"),
+            Sequence(
+                Ref.keyword("TABLE", optional=True),
+                Ref("TableReferenceSegment"),
+            ),
+            Sequence(
+                # TODO is there a less brittle way to do this using an existing class(es)?
+                # TODO if not, how to best construct a new class for `SELECT` match?
+                "FROM",
+                Ref("TableReferenceSegment"),
+                "SELECT",
+                Delimited(
+                    Ref("ColumnReferenceSegment"),
+                ),
+                Ref("TableReferenceSegment", optional=True),
```

Comment:
I don't think this should be here should it? If the table we're selecting from is before the FROM then it shouldn't also be after should it?

```suggestion
```

Reply:
Although redundant and confusing, this is valid syntax in spark. A statement with the table referenced after columns will run fine in an `insert` statement

Reply:
Some SQL dialects are just weird :-)